pool.go

```diff
@@ -71,6 +56,9 @@ type pool struct {
 	maxConns int
 	minConns int
```

Comment:
I feel a little bit uneasy because these aren't shared. But they can't be modified, right?

Reply:
By shared do you mean publicly accessible? The user can not modify the pool's min/max connection values after connecting the pool. The pool options implementation is not final. That will be revisited when we have the connection options strategy worked out.

Reply:
No, by "shared" I mean shared between multiple copies created by `withSomething` method. If they can't be modified, its fine to copy.